src/scroll-animation.js

```diff
@@ -117,7 +129,7 @@ export default class ScrollAnimation extends Component {
 
   render() {
     return (
-      <div ref={(node) => { this.node = node; }} className={this.state.classes} style={this.state.style}>
+      <div ref={(node) => { this.node = node; }} className={this.state.serverSide ? "" : this.state.classes} style={this.state.serverSide ? {} : this.state.style}>
```

Comment:
Is this bit needed?
I've just been trying your code with nextJS, and I get warnings in the browser:

"React attempted to reuse markup in a container but the checksum was invalid...."

I'm pretty unfamiliar with nextJS and serverside rendering as a whole, but googling led me [here](https://stackoverflow.com/questions/46369382/reactjs-react-attempted-to-reuse-markup-in-a-container-but-the-checksum-was-inv)

It doesn't like that that server-side and client-side markup are different. For me, the problem was fixed by going back to:
``` 
<div ref={(node) => { this.node = node; }} className={this.state.classes} style={this.state.style}>
```
But i'm unsure if you have errors with this set up in some other serverside rendering setup?

Reply:
I understand the warning and the reason behind it and there're two things we can do here. The problem is, if I remember correctly,  that if you leave the current styles and classes, part of the animation will run regardless of the scroll function, making it look broken. Of course what nextJS does is render the server version and then switch it once it has loaded react in the browser. If the markup is the same then the change will be seamless, if the animation styles change then it will jitter. Let me take a look at this tomorrow and follow up

Reply:
Ah ok. I think I understand. I can't seem to reproduce the jitter though.

The only class that gets appended to the element on serverside is 'animated', which doesn't seem to do anything on its own. The other classes (animateIn and animateOut) only get added as part of the handleScroll function right? Which is only executed via the listener, which as of your change is only added on client side execution?

I'm probably missing something. Thanks for looking into it